### PR TITLE
[2201.7.2-stage] Cleanup memory usage of ConstantValueResolver

### LINKED ISSUE #41207: [Bug]: Language Server crashes
### Description

$subject on the project https://github.com/wso2-enterprise/choreo-logging/tree/dataplane with the following stack trace

```
Active the Ballerina VS Code extension.
bal command stdout: Ballerina 2201.7.0 (Swan Lake Update 7)
Language specification 2023R1
Update Tool 1.3.15
'bal' executor is picked up by the plugin.
Ballerina version output: Ballerina 2201.7.0 (Swan Lake Update 7)
Language specification 2023R1
Update Tool 1.3.15
Plugin version: 4.4.1
Ballerina version: 2201.7.0 (Swan Lake Update 7)
Using Ballerina CLI command 'bal' for Language server.
LS offline source compilation set to true
Failed to activate Ballerina extension. command 'ballerina.view.architectureView' already exists
[Error - 13:09:11] Operation 'textDocument/semanticTokens/full' failed! {uri: '/Volumes/USERDATA/CHOREO/choreo-logging/service.bal', error: 'java.lang.NullPointerException, org.wso2.ballerinalang.compiler.tree.expressions.BLangNamedArgsExpression.accept(BLangNamedArgsExpression.java:70)'}
java.lang.NullPointerException
    at io.ballerina.compiler.api.impl.ReferenceFinder.addIfSameSymbol(ReferenceFinder.java:1437)
    at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:1041)
    at org.wso2.ballerinalang.compiler.tree.expressions.BLangNamedArgsExpression.accept(BLangNamedArgsExpression.java:70)
    at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:242)
    at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:247)
    at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:863)
    at org.wso2.ballerinalang.compiler.tree.expressions.BLangInvocation.accept(BLangInvocation.java:128)
    at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:242)
    at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:459)
    at org.wso2.ballerinalang.compiler.tree.statements.BLangExpressionStmt.accept(BLangExpressionStmt.java:51)
    at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:242)
    at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:394)
    at org.wso2.ballerinalang.compiler.tree.statements.BLangBlockStmt.accept(BLangBlockStmt.java:65)
    at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:242)
    at io.ballerina.compiler.api.impl.ReferenceFinder.findReferences(ReferenceFinder.java:233)
    at io.ballerina.compiler.api.impl.BallerinaSemanticModel.getReferences(BallerinaSemanticModel.java:289)
    at io.ballerina.compiler.api.impl.BallerinaSemanticModel.references(BallerinaSemanticModel.java:261)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.processSymbols(SemanticTokensVisitor.java:621)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:191)
    at io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode.accept(SimpleNameReferenceNode.java:41)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:421)
    at io.ballerina.compiler.syntax.tree.NamedArgumentNode.accept(NamedArgumentNode.java:49)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:152)
    at io.ballerina.compiler.syntax.tree.FunctionCallExpressionNode.accept(FunctionCallExpressionNode.java:53)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visit(NodeVisitor.java:91)
    at io.ballerina.compiler.syntax.tree.ExpressionStatementNode.accept(ExpressionStatementNode.java:45)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visit(NodeVisitor.java:79)
    at io.ballerina.compiler.syntax.tree.BlockStatementNode.accept(BlockStatementNode.java:49)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visit(NodeVisitor.java:103)
    at io.ballerina.compiler.syntax.tree.IfElseStatementNode.accept(IfElseStatementNode.java:54)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visit(NodeVisitor.java:323)
    at io.ballerina.compiler.syntax.tree.FunctionBodyBlockNode.accept(FunctionBodyBlockNode.java:58)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:137)
    at io.ballerina.compiler.syntax.tree.FunctionDefinitionNode.accept(FunctionDefinitionNode.java:66)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:233)
    at io.ballerina.compiler.syntax.tree.ServiceDeclarationNode.accept(ServiceDeclarationNode.java:82)
    at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:966)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.getSemanticTokens(SemanticTokensVisitor.java:107)
    at org.ballerinalang.langserver.semantictokens.SemanticTokensUtils.getSemanticTokens(SemanticTokensUtils.java:64)
    at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$semanticTokensFull$16(BallerinaTextDocumentService.java:665)
    at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
    at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
    at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
    at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
    at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
    at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
    at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

[Error - 13:18:13] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:13] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:13] Delivering pending changes failed
TypeError: this.task is not a function
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:1132693
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:14] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:15] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:15] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:15] Stopping server timed out
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Stopping server failed
Error: Stopping the server timed out
    at /Users/tharinduwijewardane/.vscode/extensions/wso2.ballerina-4.4.1/dist/extension.js:2:982973
[Error - 13:18:15] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:15] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:16] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:16] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:16] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:16] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:17] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:17] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:17] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:17] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:17] Client Ballerina LS Client: connection to server is erroring. Shutting down server.
[Error - 13:18:17] Client Ballerina LS Client: connection to server is erroring. Shutting down server.

```

### Steps to Reproduce

_No response_

### Affected Version(s)

2201.7.0

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose

Fixes #41207 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java [hunk 1]
```diff
@@ -579,6 +587,7 @@ BLangConstantValue constructBLangConstantValueWithExactType(BLangExpression expr
 
         this.anonTypeNameSuffixes = anonTypeNameSuffixes;
         updateConstantType(constantSymbol, expression, env);
+        reset();
```
Comment (rdulmina) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java:590]: Can u briefly explain why we need to reset here insted of calling at the end of `resolve` method?

Comment (prakanth97) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java:590]: After the symbolEnter revamp we replace the usage of constantValueResolver for constants with constantTypeChecker. So, constant annotations are only using constantValueResolver. The method `constructBLangConstantValueWithExactType` is invoked in the `symbolResolver` when resolving the const-annotation. Hence I add a reset() end of that method. Actually `resolve` method is not used at all. We have to replace the usage of this class with `constantTypeChecker` as well.

**Review by rdulmina — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java (hunk 1 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java:590]: Got it shall we call `reset()` method in https://github.com/ballerina-platform/ballerina-lang/blob/4e7e1f4d3c81a0acff294404c39b065efbc54da4/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java#L137 as well and remove single clear statements

Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java:590]: Also let's rename the method to something like `clearTempData` or similar

**Pull request merged**